app/src/main/java/com/example/pawsibilities/fragments/EditTagDialogFragment.java

```diff
@@ -41,16 +51,74 @@ public static EditTagDialogFragment newInstance(Tag tag) {
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_edit_tag, container, false);
+        binding = FragmentEditTagBinding.inflate(inflater, container, false);
+        return binding.getRoot();
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         if (getArguments() != null) {
             this.tag = getArguments().getParcelable(KEY_TAG);
+            Location l = (Location) getArguments().get(KEY_Location);
+            this.userLocation = new ParseGeoPoint(l.getLatitude(), l.getLongitude());
         }
 
-        // TODO fill in overlay's functionality
+        ParseFile profile = tag.getPhoto();
+        if (profile != null) {
+            Glide.with(getContext()).load(profile.getUrl()).transform(new RoundedCorners(150)).into(binding.ivPhoto);
+        }
+
+        binding.tvName.setText(tag.getName());
+        binding.tvTimeAgo.setText("Updated " + tag.getRelativeTimeAgo());
+        binding.tvDistance.setText(tag.distanceFrom(userLocation) + " miles away");
```

Comment:
Quick improvement. These types of user facing messages should come from resource strings. So that they can be easily translatable to many languages. (you don't need to do this in this PR)